File: src/main/java/com/markit/api/WatermarkPositionCoordinates.java
package com.markit.api;

/**
 *
 * @author Oleg Cheban
 * @since 1.0
 */
public interface WatermarkPositionCoordinates {
    Coordinates center();
    Coordinates topLeft();
    Coordinates topRight();
    Coordinates bottomLeft();
    Coordinates bottomRight();

    final class Coordinates{
        private final int x;
        private final int y;

        public Coordinates(int x, int y) {
            this.x = x;
            this.y = y;
        }

        public int getX() {
            return x;
        }

        public int getY() {
            return y;
        }
    }
}

File: src/main/java/com/markit/api/WatermarkPositionCoordinates.kt
package com.markit.api

/**
 *
 * @author Oleg Cheban
 * @since 1.0
 */
interface WatermarkPositionCoordinates {
    fun center(): Coordinates
    fun topLeft(): Coordinates
    fun topRight(): Coordinates
    fun bottomLeft(): Coordinates
    fun bottomRight(): Coordinates
    fun tiled(): List<Coordinates>
    data class Coordinates(val x: Int, val y: Int)
}